spyder/utils/tests/test_environ.py

```diff
@@ -19,45 +18,46 @@
 from qtpy.QtCore import QTimer
 
 # Local imports
-from spyder.py3compat import PY3
+from spyder.utils.environ import get_user_environment_variables, UserEnvDialog
 from spyder.utils.test import close_message_box
-from spyder.utils.environ import clean_env
 
 
 @pytest.fixture
 def environ_dialog(qtbot):
-    "Setup the Environment variables Dialog taking into account the os."
+    "Setup the Environment variables Dialog."
     QTimer.singleShot(1000, lambda: close_message_box(qtbot))
-    if os.name == 'nt':
-        from spyder.utils.environ import WinUserEnvDialog
-        dialog = WinUserEnvDialog()
-    else:
-        from spyder.utils.environ import EnvDialog
-        dialog = EnvDialog()
+    dialog = UserEnvDialog()
     qtbot.addWidget(dialog)
 
     return dialog
 
 
+def test_get_user_environment_variables():
+    """Test test_get_user_environment_variables function"""
+    os_env = os.environ.copy()
+    for k in ["DYLD_LIBRARY_PATH", "OLDPWD"]:
+        # Remove extra keys
+        os_env.pop(k, None)
+    _os_env = sorted([k.lower() for k in os_env.keys()])
+
+    guev = get_user_environment_variables()
+    for k in ["QT_MAC_WANTS_LAYER", "prompt", "qt_d3dcreate_multithreaded"]:
+        # Remove extra keys
+        guev.pop(k, None)
+    _guev = sorted([k.lower() for k in guev.keys()])
+
+    assert _os_env == _guev
```

Comment:
@ccordoba12, I'm not entirely happy with this test. Essentially, I'm checking to see if the retrieved user environment variables are the same as `os.environ`, which should be the same; except that there are some exceptions. I fear that this is not very robust and will become a game of whack-a-mole. If you have a better recommendation, I'll be happy to consider it.